ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/Server.java

```diff
@@ -89,6 +93,10 @@ public Server(final Vertx vertx, final KsqlRestConfig config, final Endpoints en
   }
 
   public synchronized void start() {
+    start(true);
+  }
+
+  private synchronized void start(final boolean startFileWatcher) {
```

Comment:
I don't think the startFileWatcher flag is necessary if you don't interrupt the watcher thread (see previous comment). This means you can simplify things and have a simple start/stop method as before and always create a new filewatcher in start().